### PR TITLE
Feature manage users via admin api

### PR DESCRIPTION
**Pull request by alrik**
This PR introduces new endpoints for creating and listing users via admin api.

### CODE REVIEW

File: src/lib/test/mocks/repositories/user.js [hunk 1]
```diff
@@ -60,14 +60,15 @@ module.exports = {
...
-  async create({id, realmId, isOnline = false}) {
+  async create({id, realmId, properties = {}, isOnline = false}) {
     if (id === duplicateUserId) {
       throw duplicateError();
     }
 
     return userModel({
       id,
       realmId,
+      properties,
```
Comment (baranga) [inline src/lib/test/mocks/repositories/user.js:71]: Nice catch :+1:

File: src/tasks/admin/create-user.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+module.exports = ({userRepository, realmRepository}) =>
+  /**
+   * @typedef {Function} AdminTasks#createUser
+   * @param {object} args
+   * @param {string} args.realmId The realm of the user
+   * @param {string} [args.id] The custom id of the user
+   * @param {object} [args.properties] An optional hash map of additional properties.
+   * @return {Promise<Result<?UserModel>>} user
+   */
+  async ({realmId, id, properties = {}}) => {
```
Comment (baranga) [inline src/tasks/admin/create-user.js:18]: **praise:** Good to see proper nested argument documentation :heart:

File: src/tasks/admin/create-user.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        return failure(
+          createTaskError({
+            code: 'UserAlreadyExists',
+            message: 'User could not be created, since a user with the same id already exists.',
+          }),
+          error
+        );
+      }
+
+      return Promise.reject(error);
```
Comment (baranga) [inline src/tasks/admin/create-user.js:41]: **nitpick:** This is in an `async` function so returning a rejected promise is not necessary.
**suggestion:** Use slicker `throw error;` instead.

**Review by baranga — APPROVED** [review]
LGTM

File: src/tasks/admin/create-user.js (hunk 3 above)
Comment (alrik) [reply src/tasks/admin/create-user.js]: Done.

**Pull request merged**